nvm.sh

```diff
@@ -2317,6 +2381,53 @@ nvm_npmrc_bad_news_bears() {
   return 1
 }
 
+# args: path to check and transform if needed
+nvm_ensure_posix_path() {
+  [ -z "${1-}" ] && nvm_err "Need the location as first parameter" && return 1
+  local location
+  location="$1"
+
+  # If already posix path, just return it
+  [ "$(command printf '%s' "$location" | command cut -b 1)" = "/" ] && nvm_echo "$location" && return 0
+
+  # If absolute path (eg. starting with C:)
+  if command printf '%s' "$location" | command grep '^[a-zA-Z]:' >/dev/null 2>&1; then
+    # Get lower case drive letter, eg. c for C:\...
+    local letter
+    letter=$(command printf '%s' "$location" | command cut -b 1 | command tr '[:upper:]' '[:lower:]')
+    # Get upper case drive letter, eg. C for C:\...
+    local letter_upper
+    letter_upper=$(command printf '%s' "$letter" | command tr '[:lower:]' '[:upper:]')
+    local letter_prefix
+    if [ -d '/mnt/' ]; then
+      letter_prefix='/mnt'
+    elif [ -d '/drives/' ]; then
+      letter_prefix='/drives'
```

Comment:
is this info not provided by the os, meaning we must test for existence?

Reply:
Good question but I have no idea. I just experienced that C:\\ would be mapped to /c on MSYS, /mnt/c on WSL, and /drives/c and/or /mnt/c on Cygwin but I have no idea how to know that.

Reply:
I think user can even customize it if wanted, but for me it seems default values that could be good to check

Reply:
Especially if it's customizable, we should try to figure out if there's a command on windows that can give the mapping prefix.

Reply:
When I mean that it is customizable, it is like if the user start updating /etc/fstab to change the mounting point.
So yes it would be possible to better now it but it would mean calling mount. I can give it a try. But then it would be really not possible to unit test under linux because it would really look for real drive to be mounted.

Reply:
Something like that could do it I guess in fact:
grep -Ee 'C: ' /etc/mtab | awk '{ print $2 }'

Reply:
So I have added this function:
```shell
nvm_windows_drive_posix() {
  nvm_grep -iEe "$1 " /etc/mtab | command awk '{ print $2 }'
}
```
So I can override it during unit tests which allow to properly test the function nvm_ensure_posix_path without the need of creating temporary folder.
It looks much cleaner in fact.